clis/teliod/QNAP.md

```diff
@@ -0,0 +1,127 @@
+# QNAP Build
+
+To build Teliod package for QNAP devices there are two alternatives:
+
+- Run cargo build directly:
+```cargo build --verbose --target x86_64-unknown-linux-musl --package teliod --features qnap``` 
+
+- Run build script [Recommended]:
+```../../ci/build_libtelio.py build qnap x86_64 [--debug]```
+
+The build script has an additional stage at the end of the build where it creates the QNAP package.
+
+## REST API
+
+This REST API allows interaction with the Teliod daemon. It provides endpoints for managing the daemon, updating its configuration, and retrieving logs and status information.
+
+### Endpoints
+
+#### 1. **Start the Daemon**
+- **Endpoint**: `/`
+- **Method**: `POST`
+- **Description**: Starts the Teliod Daemon in the background.
+- **Request Body**: None
+- **Responses**:
+  - **200 OK**: Daemon started successfully.
+  - **400 Bad Request**: Daemon is already running.
+  - **500 Internal Server Error**: Failed to start the daemon.
+
+#### 2. **Stop the Daemon**
+- **Endpoint**: `/`
+- **Method**: `DELETE`
+- **Description**: Stops the running Teliod Daemon.
+- **Request Body**: None
+- **Responses**:
+  - **200 OK**: Daemon stopped successfully.
+  - **400 Bad Request**: Daemon is not running.
+  - **500 Internal Server Error**: Failed to stop the daemon.
+
+#### 3. **Update Configuration**
+- **Endpoint**: `/`
+- **Method**: `PATCH`
+- **Description**: Updates the daemon configuration with provided settings.
+- **Request Body**: JSON object containing the configuration updates. Only specified fields will be updated; others remain unchanged.
+- **Example Request Body**:
+  ```json
+  {
+    "log_level": "info",
+    "log_file_path": "/new/path/to/log.log"
+  }
+  ```
+  - **Responses**:
+  - **200 OK**: Configuration updated successfully
+  - **400 Bad Request**: Invalid JSON payload or configuration fields.
+  - **500 Internal Server Error**: Failed to update configuration.
+
+#### 4. **Get Meshnet Status**
```

Comment:
For ui, i will probably will also need get-config, as persisting it in browser is probably not a good idea. It should return current config of teliod